lang/syn/src/codegen/program/cpi.rs

```diff
@@ -93,7 +102,11 @@ pub fn generate(program: &Program) -> proc_macro2::TokenStream {
                             &ix,
                             &acc_infos,
                             ctx.signer_seeds,
-                        ).map_err(Into::into)
+                        ).map_or_else(
+                            |e| Err(Into::into(e)),
+                            // Maybe handle Solana return data.
+                            |_| { #result_handler }
+                        )
```

Comment:
Is there a clean way to lazily deserialize the return data? Maybe A wants to cpi B but isnt interested in B's return data. I suppose one way is to make a CPI call return `anchor_lang::Result<Return<T>>` where `Return` has a `get` method that calls `get_return_data` and then does `T:try_from_slice`.

Another way is to not have CPIs have return values at all and use the normal `get_return_data` syscall (probably with an anchor wrapper function around it)

Reply:
That is an option. I like the simplicity of being able to do `let result = <cpi_call>`. It's pretty intuitive for developers. Similar to the discussion below, I think most CPI functions that return things will have a useful return and in most cases the developer will want the return data anyway. I don't have a strong handle on the performance implication though.

Reply:
I'm happy to do this work if this is the direction we want to go, the above is just my thoughts on it.

Reply:
hey sorry I forgot to answer here! feel free to just ping me on discord if I take unusually long.

yes I like `let result = <cpi_call>` much more than using `get_return_data` too

As far as lazy (de)serialization (returning `Option<>` and pulling with `.get()`) goes, if it's not unreasonably much work, Id like to just have them included with this PR. Just so the core team doesn't have to spend time revisiting this and can focus on other things. I feel like it's not a lot of added complexity in the code (but correct me if im wrong) and it's worth if we don't have to think about it anymore.

Reply:
>  if it's not unreasonably much work, Id like to just have them included with this PR. Just so the core team doesn't have to spend time revisiting this and can focus on other things.

It's not much work but this is not a good reason for anyone to implement anything or a good way to make external contributors feel valued in an open source project.

Reply:
youre right. that wasnt a good reason.

agree there's no pressing reason to have `Option<>` be included. lets wait for use cases like you said.

I moderately think lazy `get_return_data` is still a good idea. `Option<>` is something that the program writer, not the program integrator controls. I'm thinking people might gravitate towards slapping a return on anything that might be remotely useful, just cause the API for doing it is so pleasant and similar to normal rust (and tbh, maybe that is actually the right thing to do cause you cant know who might integrate with you and in what ways?).

Reply:
> I'm thinking people might gravitate towards slapping a return on anything that might be remotely useful

Yea I think you are right, I'll add the lazy return 👍

Reply:
Added